docs: language SDK page updates (#2851)
grafana/pyroscope: docs/sources/configure-client/language-sdks/dotnet.md

# .NET

Our .NET Profiler is a powerful tool designed to enhance the performance analysis and optimization of .NET applications. It seamlessly integrates with Pyroscope, offering real-time insights into the resource usage and bottlenecks within your .NET codebase. This integration empowers developers to pinpoint inefficiencies, improve application speed, and ensure resource-efficient operation.

> ### Supported Profiling Types

Reviewer: We try to use sentence case as our style, see https://grafana.com/docs/writers-toolkit/write/style-guide/capitalization-punctuation/
```suggestion
### Supported profiling types
```

The .NET Profiler supports the following profiling types:

* CPU
* Wall time
* Allocations
* Lock contention
* Exceptions

### Compatibility

Our .NET profiler works with the following .NET versions:
* .NET 6
* .NET 7
* .NET 8

> ## Before you Begin
> 
> ### Set Up a Pyroscope Server
> 
> To capture and analyze profiling data, set up a Pyroscope server. This can be:
> 
> * A **local server** for development, or
> * A **remote server** for production use.
> 
> For installation instructions, see our [Get Started]({{< relref "../../get-started" >}}) guide.
> 
> 

Reviewer: The before you begin section details the prerequisites, which you have. It doesn't seem like you need the heading for set up a Pyroscope server since that's stated in your first sentence. 

I'm curious about the mention of Cloud Profiles. Is the real prereq that a user needs to have a profiling data storage and they can use either server or cloud to do that? If so, we might want to add a sentence immediately after the heading along the lines of: "To capture and analyze profiling data, you need either a Pyroscope server or Grafana Cloud Profiles(requires a free Grafana Cloud account.) And then go into additional details. 

```suggestion
## Before you Begin

To capture and analyze profiling data, you need either a Pyroscope server or Grafana Cloud Profiles(requires a free Grafana Cloud account.) 

### Pyroscope server

The Pyroscope server can be:

* A **local server** for development, or
* A **remote server** for production use.

For installation instructions, see our [Get Started]({{< relref "../../get-started" >}}) guide.
```

> ### Using Grafana Cloud Profiles

Reviewer: If the above is correct, then we don't need Using in the heading here. 
```suggestion
### Grafana Cloud Profiles
```

Reviewer 2: Why is this better to remove the using here? I prefer the more clear title over the generic one. If it's just "grafana cloud profiles", then I'm left wondering -- _what about_ grafana cloud profiles? also for seo purposes someone would more likely search "how to use" or "using"

Reviewer: This section is before you begin -- which discusses requirements for using the procedures on the page. "Using" or "Use" implies there are steps to be followed, which we're not using. We're stating you have to have either Pyroscope server OR Grafana Cloud Profiles to use the procedure. 

For more information: https://grafana.com/docs/writers-toolkit/structure/topic-types/task/
Also, stylistically we use the imperative instead of a gerund in headings.

Reviewer: Actually, if I was going to re-do this, I'd make it much shorter. 

To capture and analyze profiling data, you need either a hosted Pyroscope OSS server or hosted Pryoscope instance with Grafana Cloud Profiles](/products/cloud/profiles-for-continuous-profiling/) (requires a free Grafana Cloud account). 

The Pyroscope server can be a local server for development or a remote server for production use.f

and that's it. Technically, the description of Grafana Cloud Profiles is not needed to understand how to do the procedure. People can use the link if they want to learn more about it. It's out of scope for the topic.

Grafana Cloud Profiles is a hosted Pyroscope service. It provides a fully managed Pyroscope server, so you don't have to worry about installing and maintaining your own server. It also provides a hosted Grafana instance for visualizing your profiling data. For more information, see [Grafana Cloud Profiles](/products/cloud/profiles-for-continuous-profiling/).

> <!-- TODO: add a section like "Learn more about reading flamegraphs and using our product" once it's ready -->

Reviewer: We have content for flamegraphs, see this PR: https://github.com/grafana/pyroscope/pull/2855

This content will be merged shortly, so you can put links to it if you want

Reviewer: ```suggestion
To learn more about using the Pyroscope UI and flamegraphs, refer to [Ingest and analyze profile data]({{< relref "../../../ingest-and-analyze-profile-data" >}}).
```

LD_PRELOAD=Pyroscope.Linux.ApiWrapper.x64.so
```

&nbsp;

### .NET Profiler API

With a managed helper you can interact with the Pyroscope profiler from .NET runtime. You can add labels, turn on/off profiling types, and more.

To use it, first, add the Pyroscope dependency:

```shell
dotnet add package Pyroscope
```

&nbsp;

### Add profiling labels to your application

> It is possible to add labels to the profiling data. These labels can be used to filter the data in the UI. Common labels include things like:

Reviewer: ```suggestion
You can add labels to the profiling data to filter the data in the UI. Common labels include:
```

Here is a simple [example](https://github.com/grafana/pyroscope/blob/main/examples/dotnet/rideshare/example/Program.cs) exposing this APIs as an http endpoint.

> ### Configuration Options

Reviewer: ```suggestion
### Configuration options
```
